src/coreclr/jit/compiler.h

```diff
@@ -5129,6 +5129,30 @@ class Compiler
                                                             GenTree*    dereferencedAddress,
                                                             InlArgInfo* inlArgInfo);
 
+    typedef JitHashTable<CORINFO_METHOD_HANDLE, JitPtrKeyFuncs<struct CORINFO_METHOD_STRUCT_>, CORINFO_METHOD_HANDLE> HelperToManagedMap;
+    HelperToManagedMap* m_helperToManagedMap = nullptr;
+
+public:
+    HelperToManagedMap* GetHelperToManagedMap()
+    {
+        if (m_helperToManagedMap == nullptr)
+        {
+            m_helperToManagedMap = new (getAllocator()) HelperToManagedMap(getAllocator());
+        }
+        return m_helperToManagedMap;
+    }
+    bool HelperToManagedMapLookup(CORINFO_METHOD_HANDLE helperCallHnd, CORINFO_METHOD_HANDLE* userCallHnd)
+    {
+        if (m_helperToManagedMap == nullptr)
+        {
+            return false;
+        }
+        bool found = m_helperToManagedMap->Lookup(helperCallHnd, userCallHnd);
+        return found;
+    }
+private:
```

Comment:
This `Map<helperMethodHnd, realManagedMethodHnd>` mapping is done purely to simplify the existing logic that tries to match `call->IsHelperCall(this, CORINFO_HELP_ARRADDR_ST)` for e.g. an optimization.

I could just do it via `call->IsHelperCall(this, CORINFO_HELP_ARRADDR_ST) || call->IsSpecialIntrinsic(this, NI_...)` - but it's more changes + we might want to allow inlining for some other helper calls as well.